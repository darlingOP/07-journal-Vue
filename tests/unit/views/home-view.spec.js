import { shallowMount } from "@vue/test-utils";
import HomeView from '@/views/HomeView'

describe('pruebas en el About View', () =>{
  test('debe concordar con el snapshot', () => {
    
    const wrapper = shallowMount(HomeView)

    expect(wrapper.html()).toMatchSnapshot()
  });  

  test('hacer click debe de redireccionar a no-entry', () => {
    const mockRouter ={
      push:jest.fn()
    }

    const wrapper = shallowMount(HomeView, {
      global:{
        mocks:{
          $router: mockRouter
        }
      }
    })

    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({name:'no-entry'})
  });
})