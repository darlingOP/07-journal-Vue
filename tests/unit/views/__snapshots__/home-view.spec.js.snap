// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el About View debe concordar con el snapshot 1`] = `
<div><img alt="Vue logo" src="">
  <h1>Hola mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
